Add plugin: Timekeep

jacobtread opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/jacobtread/obsidian-timekeep


## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "Time tracking within Obsidian",](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/manifest.json#L6)
This should not contain the word "Obsidian", it should be obvious that this is for Obsidian.

["minAppVersion": "0.15.0",](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

[import moment from "moment";](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/schema.ts#L1)
Use `import {moment} from "obsidian";` instead.

[import { Moment } from "moment";](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/timekeep.ts#L5)
And this should be `import type { Moment } from "moment";`

[this.containerEl.createEl("h2", { text: "Timekeep Settings" });](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L15)
[Don't include a header with the plugin name in the settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[.setName("Timestamp Display Format")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L18), [.setName("PDF Title")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L41), [.setName("CSV Heading Row")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L55), [.setName("CSV Delimiter")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L68), [.setName("Timestamp Durations")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L83), [.setName("Display Segments in Reverse Order")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L96), [.setName("Limit Table Height")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L109)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const styles = StyleSheet.create({](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/components/TimesheetPdf.tsx#L16-L112), [style={{](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/components/TimesheetRow.tsx#L59-L65)
This should be in the CSS file so themes & snippets can actually change it.

jacobtread commented:
> ["description": "Time tracking within Obsidian",](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/manifest.json#L6)
This should not contain the word "Obsidian", it should be obvious that this is for Obsidian.

I have removed this from the description

> ["minAppVersion": "0.15.0",](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

I have updated this to match the latest app version

> [import moment from "moment";](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/schema.ts#L1)
Use import {moment} from "obsidian"; instead.

> [import { Moment } from "moment";](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/timekeep.ts#L5)
And this should be import type { Moment } from "moment";

This is not applicable as moment is imported directly from the moment package because it's used in unit tests which do not have access to the obsidian package so this is important separately to maintain testability

> [.setName("Timestamp Display Format")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L18), [.setName("PDF Title")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L41), [.setName("CSV Heading Row")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L55), [.setName("CSV Delimiter")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L68), [.setName("Timestamp Durations")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L83), [.setName("Display Segments in Reverse Order")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L96), [.setName("Limit Table Height")](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/settings-tab.ts#L109)

These have been updated to use sentence case

> [const styles = StyleSheet.create({](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/components/TimesheetPdf.tsx#L16-L112), 

StyleSheet.create is not using standard CSS, and is not expected to match the obsidian theme. This is the special styling for react-pdf which is being used to generate pdfs which are *not* styled the same as Obsidian UI

> [style={{](https://github.com/jacobtread/obsidian-timekeep/blob/8b36856cdd034a7caaf83f3ea6c052841e45d374/src/components/TimesheetRow.tsx#L59-L65)

I have moved this to styles.css
